gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/util/policy/ServicePolicyXMLSimplifier.java

```diff
@@ -0,0 +1,41 @@
+package com.ca.apim.gateway.cagatewayexport.util.policy;
+
+import com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentParseException;
+import org.w3c.dom.Element;
+
+import javax.inject.Singleton;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.ca.apim.gateway.cagatewayconfig.util.policy.PolicyXMLElements.*;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.getSingleChildElement;
+import static com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentUtils.getSingleElement;
+
+
+@Singleton
+public class ServicePolicyXMLSimplifier {
+    private static final Logger LOGGER = Logger.getLogger(ServicePolicyXMLSimplifier.class.getName());
+    /**
+     *
+     * @param policyElement service policy element.
+     * @return true if ApiPortalIntegration assertion is present and PortalManagedApiFlag is ApiPortalManagedServiceAssertion.
+     */
+    public String simplifyServicePolicyXML(Element policyElement) {
+        Element portalIntegrationElement = null;
+        String portalManagedApiFlag = null;
+        try {
+            portalIntegrationElement = getSingleElement(policyElement, API_PORTAL_INTEGRATION);
+        } catch (DocumentParseException e) {
+            LOGGER.log(Level.INFO, "ApiPortalIntegration assertion is not found in service policy");
+        }
+
+        if (portalIntegrationElement != null) {
+            Element portalManagedApiFlagElement = getSingleChildElement(portalIntegrationElement, PORTAL_MANAGED_API_FLAG, true);
+            if(portalManagedApiFlagElement != null){
+                portalManagedApiFlag = portalManagedApiFlagElement.getAttribute(API_PORTAL_SERVICE_ASSERTION);
+                portalIntegrationElement.removeChild(portalManagedApiFlagElement);
```

Comment:
Do we have to remove portal assertion, it has information about api key and group ?

Reply:
Good question. Let me check.

Reply:
@uppoju  

Looking at the implementation of "Set as Portal Managed Service" server assertion (modules\gateway\assertions\ApiPortalIntegrationAssertion\src\main\java\com\l7tech\external\assertions\apiportalintegration\server\ServerApiPortalIntegrationAssertion.java),

Ideally, it is used as marker assertion but in this case, it is doing something extra. It was not the case with marker assertion used for ENCAS. 

That means, during export when migration is enabled, we should replace it with the equivalent SET-assertions.

In addition to this, please consider commenting/disabling the MARKER assertion instead of removing it. Do this for both ENCAS and SERVICE.